javascript/src/stores/streams/StreamsStore.ts

```diff
@@ -16,16 +14,116 @@ interface Stream {
     createdAt: number;
 }
 
-var StreamsStore = {
-    listStreams(): JQueryPromise<Array<Stream>> {
-        var url = jsRoutes.controllers.api.StreamsApiController.listStreams().url;
-        var promise = $.getJSON(url);
-        promise.fail((jqXHR, textStatus, errorThrown) => {
-            UserNotification.error("Loading streams failed with status: " + errorThrown,
-                "Could not load streams.");
-        });
-        return promise;
-    },
+interface TestMatchResponse {
+    matches: boolean;
+    rules: any;
+}
+
+interface Callback {
+    (): void;
+}
+
+class StreamsStore {
+    private callbacks: Array<Callback> = [];
+
+    listStreams() {
+        return $.getJSON(jsRoutes.controllers.api.StreamsApiController.list().url);
```

Comment:
The method is used in other places of the code which are outside of the scope of this PR. It was introduced during the search page overhaul and I left it in to be keep this PR as nonintrusive as possible. It is not doing any error handling because it's just returning the promise. We should refactor the code which is using this though, but this is beyond this PR IMO.